### PR TITLE
IA-2194 Migrate to 2.13

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-2194

* Lots of diff is due to `scala.jdk.CollectionConverters._`
* Migrating to akka-http 10.2 caused some deprecation as well

Validation:
1. All 65 automation tests ran
2. Tested manually with notebook functions, and works fine
---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala:97]: see https://doc.akka.io/docs/akka-http/current/migration-guide/migration-guide-10.2.x.html#configuring-https-connections

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala
Comment (kyuksel) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala:128]: I wonder why `Authorization` was in quotes before.

**Review by kyuksel — APPROVED** [review]
LGTM 👍  May be worth running the integration tests a few more times (once they pass) in case some tricky flakiness lies beneath. 

Thanks for pushing this through!

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala
Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala:128]: I think `(backtick)Authorization(backtick)` was the literal class name before. I'm going to have to get used to `foo(Type)` instead of `foo[Type]`

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala:309]: I had thought attributes were different than headers? Is this an akka recommended change?

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/CommonTestData.scala
Comment (rtitle) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/CommonTestData.scala:94]: Is this needed?

**Comment by rtitle** [comment]
Looks good, thanks for doing this! Looking at Jenkins, seems there are some sbt `assembly` failures.

Agree with @kyuksel about running the tests a few times.. may also be good to do some manual smoke testing (e.g. running a notebook) to make sure no unexpected issues with Akka+websockets.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala
Comment (kyuksel) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala:128]: ugh indeed...`foo(Type)` is so confusing 😕

**Comment by Qi77Qi** [comment]
need most recent google2...blocked by https://github.com/DataBiosphere/leonardo/pull/1679

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Review by rtitle — APPROVED** [review]
This still looks good to me, if tests are passing reliably.

By the way, I had to make the `ApplicativeAsk` -> `Ask` change in one of my branches because I updated to the latest `google2`. But it's all localized to 1 commit, so I can drop the commit and rebase once this PR is merged.

**Comment by Qi77Qi** [comment]
```      org.scalatest.exceptions.TestFailedException: 2020-11-17T20:50:13.611Z was not less than 2020-11-17T20:50:13.611Z
      at org.scalatest.matchers.MatchersHelper$.indicateFailure(MatchersHelper.scala:339)
```
Looks like flaky test

jenkins retest

**Pull request merged**
